File: tools/list_icons.py
import pathlib
import re

repo_root = pathlib.Path(__file__).resolve().parents[1]
root = repo_root / 'loanflow' / 'src'
if not root.exists():
    root = pathlib.Path('loanflow/src')
patterns = [
    r'name="([^"]+)"',
    r"name='([^']+)'",
    r'\[(?:name|icon)\]="([^"]+)"',
    r"\[(?:name|icon)\]='([^']+)'",
    r'icon="([^"]+)"',
    r"icon='([^']+)'",
]
values = set()
for path in root.rglob('*'):
    if path.suffix.lower() not in {'.ts', '.html'}:
        continue
    text = path.read_text(encoding='utf-8', errors='ignore')
    for pattern in patterns:
        values.update(re.findall(pattern, text))

values = sorted(v for v in values if v and not v.strip().startswith('{'))
print('\n'.join(values))

File: tools/list_icons.mjs
import fs from 'fs';
import path from 'path';

const repoRoot = path.resolve(path.dirname(new URL(import.meta.url).pathname), '..');
let searchRoot = path.join(repoRoot, 'loanflow', 'src');
if (!fs.existsSync(searchRoot)) {
  searchRoot = path.join('loanflow', 'src');
}

const tagPattern = /<ion-icon[^>]*?(?:\[(?:name|icon)\]|(?:name|icon))\s*=\s*("([^"]+)"|'([^']+)')/gi;
const tsPattern = /addIcons\(([^)]+)\)/g;
const stringPattern = /'([^']+)'|"([^"]+)"/g;

const iconNamePattern = /^[a-z][a-z0-9-]*$/i;
const values = new Set();

function walk(dir) {
  const entries = fs.readdirSync(dir, { withFileTypes: true });
  for (const entry of entries) {
    const fullPath = path.join(dir, entry.name);
    if (entry.isDirectory()) {
      walk(fullPath);
    } else if (/\.html$/i.test(entry.name)) {
      const content = fs.readFileSync(fullPath, 'utf8');
      let match;
      while ((match = tagPattern.exec(content)) !== null) {
        const value = match[2] ?? match[3];
        if (value && iconNamePattern.test(value)) {
          values.add(value);
        }
      }
    } else if (/\.ts$/i.test(entry.name)) {
      const content = fs.readFileSync(fullPath, 'utf8');
      let match;
      while ((match = tagPattern.exec(content)) !== null) {
        const value = match[2] ?? match[3];
        if (value && iconNamePattern.test(value)) {
          values.add(value);
        }
      }

      tsPattern.lastIndex = 0;
      while ((match = tsPattern.exec(content)) !== null) {
        const block = match[1];
        let inner;
        stringPattern.lastIndex = 0;
        while ((inner = stringPattern.exec(block)) !== null) {
          const value = inner[1] ?? inner[2];
          if (value && iconNamePattern.test(value)) {
            values.add(value);
          }
        }
      }
    }
  }
}

walk(searchRoot);

console.log(Array.from(values).sort().join('\n'));
